### PR TITLE
[KRAKEN-97] fix(requestTimeTooSkewed) Try a fix clock in aws config

### PR DESCRIPTION
**Pull request by MatthD**
## 📝 What is this PR

**Explain what is this PR Fix/Add, screenshots are welcome**

-  [SPEC](https://phantombuster.atlassian.net/browse/KRAKEN-97)

## 🏭 How does it work

**Explain what seems important inside your code for reviewer**

- Force the `correctClockSkew`

## 🧪 How to test ?

- To replicate **go to Pakistan** ... or just set it's timezone (+4hours , France ref) on you computer
- Start front/back locally and navigate to file browser 
Capture d’écran 2021-03-23 à 14 54 35

You should come to an error (`time too much different ...` or something else)

- Now update the `pb2-front` temp for testing the current version inside `src/pages/Agent/index.tsx` set the **url** variable to `const url = `https://deploy-preview-11--jolly-joliot-aafb20.netlify.app/?settings=${btoa(JSON.stringify(s3BrowserSettings))}`
- Restart front & retry
- It should work 🙏

https://user-images.githubusercontent.com/7901366/112158474-7bc34380-8be8-11eb-8c1d-3448c2883ad8.mov

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,254 +1,39 @@
...
-* bucket-name.s3-website.region.amazonaws.com
+## Start it locally
 
-Note the dash (-) between s3-website and the region identifier. Which form is used for the endpoint depends on what Region the bucket is in. For a complete list, please see [Amazon S3 Website Endpoints](https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_website_region_endpoints).
+- `cd aws-js-s3-explorer`
+- `npx http-server . 8081`
+- access it at [localhost:8081](localhost:8081)
 
-## Display Options
+## Link to it in front locally
```
Comment (YungVDev) [inline README.md]:
```suggestion
## Edit pb2-front to target the local aws explorer

```

Or maybe just:
```suggestion
## Test locally

```

File: README.md [hunk 2]
```diff
@@ -1,254 +1,39 @@
...
 
-This application allows visitors to view the contents of a bucket via its folders or by listing out all objects in a bucket. The default view is by folder, but users can choose Initial View: Bucket in Settings to display all objects in the bucket. Note that viewing an entire bucket that contains many objects could overwhelm the browser. We&rsquo;ve successfully tested this application on a bucket with over 30,000 objects, but keep in mind that trying to list too many objects in a browser could lead to a poor user experience.
+- `cd pb2-front`
+- edit `src/pages/Agent/index.tsx`
+- set `const url = localhost:8081/?settings=${btoa(JSON.stringify(s3BrowserSettings))}`
+- Start/Restart front
 
-## Credentials
+## Test in in stagging
```
Comment (YungVDev) [inline README.md]:
```suggestion
## Test in staging

```

File: README.md [hunk 3]
```diff
@@ -1,254 +1,39 @@
...
+- `npx http-server . 8081`
+- access it at [localhost:8081](localhost:8081)
 
-## Display Options
+## Link to it in front locally
 
-This application allows visitors to view the contents of a bucket via its folders or by listing out all objects in a bucket. The default view is by folder, but users can choose Initial View: Bucket in Settings to display all objects in the bucket. Note that viewing an entire bucket that contains many objects could overwhelm the browser. We&rsquo;ve successfully tested this application on a bucket with over 30,000 objects, but keep in mind that trying to list too many objects in a browser could lead to a poor user experience.
+- `cd pb2-front`
+- edit `src/pages/Agent/index.tsx`
+- set `const url = localhost:8081/?settings=${btoa(JSON.stringify(s3BrowserSettings))}`
```
Comment (YungVDev) [inline README.md:25]: Maybe specify in which function we should set this line?

File: README.md [hunk 4]
```diff
@@ -1,254 +1,39 @@
...
 
-## Credentials
+## Test in in stagging
 
-This tool can be used for publicly-readable S3 buckets (where no authentication is required) and for private S3 buckets (where authentication *is* required).
+- First open a PR pointing to the `prod` branch ☢️ It will automatically point to the original fork repository, first change to phantombuster then point to prod branch 🙏 ☢️
+- After a while a netlify link will appear inside the PR check details, just click on `details` button to open the stagging test & copy the url
+- `cd pb2-front`
+- edit `src/pages/Agent/index.tsx`
+- set `const url = ${url-to-generated-netlify-s3}:8081/?settings=${btoa(JSON.stringify(s3BrowserSettings))}`
```
Comment (YungVDev) [inline README.md:34]: Same as above, maybe specify in which function we should set this line?

File: explorer.js [hunk 5]
```diff
@@ -149,6 +149,8 @@ function SharedService($rootScope) {
 		// AWS.config.update(settings.cred);
 		// AWS.config.update({ region: settings.region });
 		AWS.config.update(Object.assign(settings.cred, { region: settings.region }));
+		AWS.config.update({ correctClockSkew: true });
+		AWS.config.correctClockSkew = true;
```
Comment (YungVDev) [inline explorer.js]: Do we really need both? Isn't `AWS.config.update({ correctClockSkew: true })` enough?

File: explorer.js [hunk 6]
```diff
@@ -1340,6 +1342,8 @@ $(document).ready(() => {
 	// Default AWS region and v4 signature
 	AWS.config.update({ region: '' });
 	AWS.config.update({ signatureVersion: 'v4' });
+	AWS.config.update({ correctClockSkew: true });
+	AWS.config.correctClockSkew = true;
```
Comment (YungVDev) [inline explorer.js]: Do we really need both? Isn't AWS.config.update({ correctClockSkew: true }) enough?

**Review by YungVDev — CHANGES_REQUESTED** [review]

File: explorer.js (hunk 5 above)
Comment (MatthD) [reply explorer.js]: the doc about it is really not clear, do not know if it's an alias on it do thing differently, both are listed in a different way, I could prefer to cover all possible cases which could happen in different scenarii (update seems to point to previous usage of config )
https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html

File: explorer.js (hunk 6 above)
Comment (MatthD) [reply explorer.js]: same as above, woule prefer the two usage listed in the doc to be sure we really fix it for all usage

File: explorer.js (hunk 5 above)
Comment (YungVDev) [reply explorer.js]: > the doc about it is really not clear, do not know if it's an alias on it do thing differently, both are listed in a different way, I could prefer to cover all possible cases which could happen in different scenarii (update seems to point to previous usage of config )
> https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html

I think `config.update` is just an alias, if you only do `AWS.config.update({ correctClockSkew: true });` and then do `console.log(AWS.config);` you will notice that AWS.config has a new `correctClockSkew` property set to `true`

File: explorer.js (hunk 6 above)
Comment (YungVDev) [reply explorer.js]: > same as above, woule prefer the two usage listed in the doc to be sure we really fix it for all usage

I think config.update is just an alias, if you only do AWS.config.update({ correctClockSkew: true }); and then do console.log(AWS.config); you will notice that AWS.config has a new correctClockSkew property set to true

**Review by YungVDev — CHANGES_REQUESTED** [review]
Seems to work based on your video in the description however I could not manage to reproduce the bug even by changing my timezone. Could you help me with that? :smile: 

Also, I think you can remove the `AWS.config.correctClockSkew = true` line to only keep the config.update

**Review by YungVDev — APPROVED** [review]
LGTM :tada:

**Review by Oxygen66 — APPROVED** [review]
LGTM 💯

**Pull request merged**